### ISSUE TITLE
Review logging of MapboxNavigationTelemetry

### ISSUE THREAD

**Issue by RingerJK** [mapbox/mapbox-navigation-android#3853]
Telemetries' TAG("MAPBOX_TELEMETRY") is used outside of `MapboxNavigationTelemetry` class, which is not right and confused.

**Comment by Guardiola31337** [comment]
Saw that this https://github.com/mapbox/mapbox-navigation-android/issues/3865 landed, are we good to close here @RingerJK? 🤔

**Comment by RingerJK** [comment]
@Guardiola31337 no, it's another problem. I linked because it's also an issue with logging
